pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -353,6 +353,29 @@ func (b *AutoscalingGroupModelBuilder) buildAutoScalingGroupTask(c *fi.ModelBuil
 
 	t.InstanceProtection = ig.Spec.InstanceProtection
 
+	// When Spotinst Elastigroups are used, there is no need to create
+	// a separate task for the attachment of the load balancer since this
+	// is already done as part of the Elastigroup's creation, if needed.
+	if !featureflag.Spotinst.Enabled() {
+		if b.UseLoadBalancerForAPI() && ig.Spec.Role == kops.InstanceGroupRoleMaster {
+			t.LoadBalancers = append(t.LoadBalancers, b.LinkToELB("api"))
+		}
+
+		if ig.Spec.Role == kops.InstanceGroupRoleBastion {
+			t.LoadBalancers = append(t.LoadBalancers, b.LinkToELB("bastion"))
+		}
+	}
+
+	for _, extLB := range ig.Spec.ExternalLoadBalancers {
+		if extLB.LoadBalancerName != nil {
+			t.LoadBalancers = append(t.LoadBalancers, &awstasks.LoadBalancer{Name: extLB.LoadBalancerName})
```

Comment:
nit: it seems a bit strange to be creating awstasks here and then creating duplicate tasks below and adding them to the builder context. Maybe there's a way we can both append and AddTask the same task objects

Reply:
Nice catch cleaned up.